src/test/ui/consts/min_const_fn/min_const_fn.rs

```diff
@@ -136,9 +136,7 @@ const fn no_rpit() -> impl std::fmt::Debug {} //~ ERROR `impl Trait` in const fn
 const fn no_dyn_trait(_x: &dyn std::fmt::Debug) {} //~ ERROR trait bounds other than `Sized`
 const fn no_dyn_trait_ret() -> &'static dyn std::fmt::Debug { &() }
 //~^ ERROR trait bounds other than `Sized`
-//~| WARNING cannot return reference to temporary value
-//~| WARNING this error has been downgraded to a warning
-//~| WARNING this warning will become a hard error in the future
+//~| ERROR cannot return reference to temporary value
```

Comment:
Note to self: min const fn qualification failing causing temporary promotion to fail really should be fixed.

Reply:
Might be worth filing an issue; cc @oli-obk

Reply:
Should probably go away once we decouple promotion and const-checking.